src/main/java/com/scaleunlimited/flinkcrawler/topology/CrawlTopologyBuilder.java

```diff
@@ -185,11 +198,20 @@ public CrawlTopologyBuilder setSiteMapParser(BasePageParser siteMapParser) {
         return this;
     }
 
-    public CrawlTopologyBuilder setContentSink(SinkFunction<ParsedUrl> contentSink) {
+    public CrawlTopologyBuilder setContentSink(SinkFunction<Tuple2<NullWritable, WARCWritable>> contentSink) {
         _contentSink = contentSink;
         return this;
     }
 
+    public CrawlTopologyBuilder setContentFile(String filePathString) {
+        if (_contentSink != null) {
```

Comment:
These kinds of "conflicting options" checks should probably be handled instead by a validate method (see CrawlToolOptions). I know I authored some of these myself here CrawlTopologyBuilder, but maybe this is a chance to clean them up.

Reply:
OK - I will try and clean them up.

Reply:
It looks like you punted on this, but I'm not going to hold up the review for it.